### PR TITLE
CodePush definition update changes in Electrode OTA Server

### PR DESCRIPTION
**Pull request by scrajesh**
CodePush definition update for the new Service
source: https://github.com/microsoft/code-push/pull/645/files

The changes here responds to new `/update_check`, `/report_status/deploy`, `/report_status/download` services, all with new set of parameters(modified the queries to camelCase to snake_case).

Still the existing `/updateCheck`, `/reportStatus/deploy`, `/reportStatus/download` services are supported as well.

### CODE REVIEW

File: electrode-ota-server-routes-acquisition/src/index.js [hunk 1]
```diff
@@ -1,48 +1,79 @@
...
         updateCheck,
         downloadReportStatus,
         deployReportStatus
-
     } = wrap(acquisition);
 
-    route([
+    const handleUpdateCheck = (request, reply) => {
+        logger.info(reqFields(request), "updateCheck request");
+        const snakeCaseRequested = /update_check/.test(request.url.path);
```
Comment (datvong-wm) [inline electrode-ota-server-routes-acquisition/src/index.js]: you'll probably want to be more specific in this test.   `/updateCheck?update_check=true` will also be true.

File: electrode-ota-server-routes-acquisition/src/index.js [hunk 2]
```diff
@@ -1,48 +1,79 @@
...
+            }
+            // if snake_case request? convert the response to the same
+            const updateInfo = snakeCaseRequested ? keysToCamelOrSnake(updatedInfo) : updatedInfo;
+            reply({ updateInfo });
+        });
+    };
+
+    const handleReportDeployStatus = (request, reply) => {
+        logger.info(reqFields(request), "report deployment status request");
+        const snakeCaseRequested = /report_status/.test(request.url.path);
```
Comment (datvong-wm) [inline electrode-ota-server-routes-acquisition/src/index.js]: same as above

File: electrode-ota-server-routes-acquisition/src/index.js [hunk 3]
```diff
@@ -1,48 +1,79 @@
...
+        // if snake_case request? convert to camelCase
+        const payload = snakeCaseRequested ? keysToCamelOrSnake(request.payload) : request.payload;
+        deployReportStatus(payload, ok(reply));
+    };
+
+    const handleReportDownloadStatus = (request, reply) => {
+        logger.info(reqFields(request), "report download status request");
+        const snakeCaseRequested = /report_status/.test(request.url.path);
```
Comment (datvong-wm) [inline electrode-ota-server-routes-acquisition/src/index.js]: same as above

File: electrode-ota-server-routes-acquisition/src/index.js [hunk 4]
```diff
@@ -1,48 +1,79 @@
...
+        // if snake_case request? convert to camelCase
+        const payload = snakeCaseRequested ? keysToCamelOrSnake(request.payload) : request.payload;
+        downloadReportStatus(request.payload, ok(reply));
```
Comment (datvong-wm) [inline electrode-ota-server-routes-acquisition/src/index.js]: how is this passing tests?

File: electrode-ota-server/test/electrode-ota-server-test.js [hunk 5]
```diff
@@ -78,5 +78,12 @@ describe('electrode-ota-server', function () {
...
-
+    it('should respond to newly updated service "/updated_check" with modified params', request({
+        url: '/update_check?deployment_key=evHVNrwuJDVdTvpSouLxKpEyowtcxyWcyVCxnsFD&app_version=1.2.3&package_hash=&is_companion=&label=&client_unique_id=76451410-77F2-423C-836C-623AA5F586B5'
+    }, body({
+        updateInfo: {
+            is_available: false,
+            should_run_binary_version: false
+        }
+    })));
 });
```
Comment (datvong-wm) [inline electrode-ota-server/test/electrode-ota-server-test.js]: need more tests

File: electrode-ota-server-routes-acquisition/src/index.js (hunk 1 above)
Comment (scrajesh) [reply electrode-ota-server-routes-acquisition/src/index.js]: using`request.url.pathname` to determine the request action

File: electrode-ota-server-routes-acquisition/src/index.js (hunk 4 above)
Comment (scrajesh) [reply electrode-ota-server-routes-acquisition/src/index.js]: updated to simply `payload` var

File: electrode-ota-server/test/electrode-ota-server-test.js (hunk 5 above)
Comment (scrajesh) [reply electrode-ota-server/test/electrode-ota-server-test.js]: added more tests

File: electrode-ota-server/test/electrode-ota-server-test.js [hunk 6]
```diff
@@ -78,5 +78,48 @@ describe('electrode-ota-server', function () {
         }
     })));
 
+    it('should respond to newly updated service "/update_check" with modified params', request({
+        url: '/update_check?deployment_key=evHVNrwuJDVdTvpSouLxKpEyowtcxyWcyVCxnsFD&app_version=1.2.3&package_hash=&is_companion=&label=&client_unique_id=76451410-77F2-423C-836C-623AA5F586B5'
+    }, body({
+        updateInfo: {
```
Comment (datvong-wm) [inline electrode-ota-server/test/electrode-ota-server-test.js]: should this be  update_info or the same updateInfo?

Comment (scrajesh) [reply electrode-ota-server/test/electrode-ota-server-test.js]: it's `updateInfo` as seen in the following link: https://github.com/microsoft/code-push/pull/645/files#diff-a937c637a47cbd31cbb52c89bef7d197R126

**Pull request merged**